code/__DEFINES/components.dm

```diff
@@ -305,6 +305,7 @@
 
 // /obj/item/clothing signals
 #define COMSIG_SHOES_STEP_ACTION "shoes_step_action"			//from base of obj/item/clothing/shoes/proc/step_action(): ()
+#define COMSIG_SUIT_SPACE_TOGGLE "suit_space_toggle"			//from base of /obj/item/clothing/suit/space/proc/toggle_spacesuit() : (obj/item/clothing/suit/space/suit)

```

Comment:
theres already a signal for action buttons being hit isnt there

Reply:
COMSIG_ACTION_TRIGGER is triggered from datum/action/proc/Trigger()

COMSIG_SUIT_SPACE_TOGGLE is triggered from /obj/item/clothing/suit/space/proc/toggle_spacesuit()

This is not a signal for the button being pressed, this was a signal that the suit changing state and the action button happens to be the listener. The reverse of COMSIG_ACTION_TRIGGER .

Reply:
Additional this is because the suit changes state separately from the action button, and the action button has to be updated correctly to prevent the suit looking on when it is really off.  This is to prevent the bug that internals has where the action button shows active when it is not active, or you are not wearing internals anymore.

Reply:
doesnt the suit have a reference to the action button? cant you direct call a proc on the button?

Reply:
That was what I did earlier and what oranges asked me to change to a signal, so I am not reverting that.

Reply:
this is cleaner than having a reference to the action button when you don't need it, just so you can update it it's icon.